addons/stock/report/report_stock_forecasted.py

```diff
@@ -107,13 +107,32 @@ def _get_report_data(self, product_template_ids=False, product_variant_ids=False
             res['virtual_available'] = sum(product_variants.mapped('virtual_available'))
         res.update(self._compute_draft_quantity_count(product_template_ids, product_variant_ids, wh_location_ids))
 
-        res['lines'] = self._get_report_lines(product_template_ids, product_variant_ids, wh_location_ids)
+        # Get the name of the order if it exists, of the picking otherwise (for SO we need to use order_line_id to avoid problem with the qty_at_date widget)
+        if self.env.context.get('order_name'):
+            order_name = self.env.context.get('order_name')
+        elif self.env.context.get('order_line_id'):
+            order_name = self.env['sale.order.line'].browse(self.env.context.get('order_line_id')).order_id.name
+        elif self.env.context.get('picking_name'):
+            order_name = self.env.context.get('picking_name')
+        else:
+            order_name = None
```

Comment:
You can't do that directly in `stock` as the models `sale.order`/`sale.order.line` could be missing.
You can do a method to get the order name and override it in `sale_stock` instead.

Also, you can use a `search_read` to find the sale order name without get the SO line first
```suggestion
        elif self.env.context.get('order_line_id'):
            sale_order = self.env['sale.order'].search_read(
                [('order_line', 'in', self.env.context['order_line_id'])],
                ['name'],
                limit=1,
            )[0]
            order_name = sale_order['name']
        else:
            order_name = self.env.context.get('picking_name', None)
```